test/e2e/traces_test.go

```diff
@@ -0,0 +1,244 @@
+//go:build experimentalintegration
+// +build experimentalintegration
+
+package e2e
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/efficientgo/e2e"
+	"github.com/efficientgo/tools/core/pkg/testutil"
+)
+
+// OTel trace collector that receives in HTTP w/o security, but exports in gRPC with security.
+const (
+	otelForwardingConfig = `
+receivers:
+    otlp:
+      protocols:
+        http:
+            endpoint: 0.0.0.0:4318
+        grpc:
+            endpoint: 0.0.0.0:4317
+
+# extensions:
+#     oauth2client:
+#       client_id: {{OBS_CLIENT_ID}}
+#       client_secret: {{OBS_CLIENT_SECRET}}
+#       token_url: {{OIDC_TOKEN_URL}}
+#      # The test dex's certs are signed for localhost, not 'e2e_traces_read_export-dex'
+#      tls:
+#        insecure_skip_verify: true
+  
+exporters:
+    logging:
+      logLevel: debug
+    otlp:
+      endpoint: {{OBS_GRPC_ENDPOINT}}
+      # auth:
+      #   authenticator: oauth2client
+      tls:
+        insecure_skip_verify: true
+      headers:
+        x-tenant: test-oidc
+        # (Use hard-coded auth header, because this forwarding collector
+        # is unable to do OIDC password grant.)
+        authorization: bearer {{DEX_TOKEN}}
+
+service:
+    telemetry:
+      metrics:
+        address: localhost:8889
+    # extensions: [oauth2client]
+    pipelines:
+      traces:
+        receivers: [otlp]
+        exporters: [logging,otlp]
+`
+	traceJSON = `
+{
+	"resource_spans": [
+	{
+		"resource": {
+		"attributes": [
+			{
+			"key": "host.name",
+			"value": { "stringValue": "testHost" }
+			}
+		]
+		},
+		"instrumentation_library_spans": [
+		{
+			"spans": [
+			{
+				"trace_id": "5B8EFFF798038103D269B633813FC60C",
+				"span_id": "EEE19B7EC3C1B173",
+				"name": "testSpan",
+				"start_time_unix_nano": 1544712660000000000,
+				"end_time_unix_nano": 1544712661000000000,
+				"attributes": [
+				{
+					"key": "attr1",
+					"value": { "intValue": 55 }
+				}
+				]
+			}
+			]
+		}
+		]
+	}
+	]
+}`
+)
+
+func TestTracesExport(t *testing.T) {
```

Comment:
This method doesn't run a test yet [as per CI](https://app.circleci.com/pipelines/github/observatorium/api/471/workflows/3f2c2901-2ee6-49bc-b80b-6a7d1c50a918/jobs/2289), the reason being that it doesn't have a `t.Run` call yet. 

I think that [this block](https://github.com/observatorium/api/pull/246/files#diff-55863b90b036d7b2b31833355b1bfaef5ad7140e9f67e7364ac710f3f2e7af81R192-R224) can be added within a `t.Run` as it is sending the trace and then querying for it, so `t.Run("write-then-query-single-trace", func..` would be a good idea! 🙂